cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -166,11 +166,15 @@ public ResourceReport getResourceReport() {
     String podName = String.format("%s-%d", meta.getName(), instanceId);
 
     try {
-      api.deleteNamespacedPodAsync(podName, kubeNamespace, null, deleteOptions, null, null, null, null,
-                                   createCallbackFutureAdapter(resultFuture, r -> runnable));
+      // TODO: CDAP-18000 Make this async call when CDAP-18000 is fixed
+      api.deleteNamespacedPod(podName, kubeNamespace, null, deleteOptions, null, null, null, null);
     } catch (ApiException e) {
       completeExceptionally(resultFuture, e);
+    } catch (Exception ex) {
+      // TODO: CDAP-18000 Ignore any exception apart from ApiException till we upgrade the kubernetes client library
+      LOG.trace("Ignoring exception while deleting pod {}", podName, ex);
```

Comment:
one question should we make this debug in case we want to know we received any exception?

Reply:
Ideally library would throw `ApiException`. We are getting exception during deserialization of response because of incompatibility between the response and expected type by the kubernetes library. Since this exception would happen always, logging it as trace to not flood logs.